packages/babel-plugin-proposal-pipeline-operator/src/smartVisitor.js

```diff
@@ -1,41 +1,47 @@
 import { types as t } from "@babel/core";
-import minimalVisitor from "./minimalVisitor";
 
 const updateTopicReferenceVisitor = {
   PipelinePrimaryTopicReference(path) {
     path.replaceWith(this.topicId);
   },
-  AwaitExpression(path) {
-    throw path.buildCodeFrameError(
-      "await is not supported inside pipeline expressions yet",
-    );
-  },
-  YieldExpression(path) {
-    throw path.buildCodeFrameError(
-      "yield is not supported inside pipeline expressions yet",
-    );
-  },
   PipelineTopicExpression(path) {
     path.skip();
   },
 };
 
 const smartVisitor = {
-  ...minimalVisitor,
-  PipelineTopicExpression(path) {
-    const topicId = path.scope.generateUidIdentifier("topic");
+  BinaryExpression(path) {
+    const { scope } = path;
+    const { node } = path;
+    const { operator, left, right } = node;
+    if (operator !== "|>") return;
 
-    path.traverse(updateTopicReferenceVisitor, { topicId });
+    const placeholder = scope.generateUidIdentifierBasedOnNode(left);
+    scope.push({ id: placeholder });
 
-    const arrowFunctionExpression = t.arrowFunctionExpression(
-      [topicId],
-      path.node.expression,
-    );
+    let call;
+    if (t.isPipelineTopicExpression(right)) {
+      path
+        .get("right")
+        .traverse(updateTopicReferenceVisitor, { topicId: placeholder });
 
-    path.replaceWith(arrowFunctionExpression);
-  },
-  PipelineBareFunction(path) {
-    path.replaceWith(path.node.callee);
+      call = right.body ? right.body : right.expression;
```

Comment:
What is this check needed for?

Reply:
Good catch!

It seems like an artifact from the refactor from minimal. Since we're dealing with a topic expression here, `body` will always be undefined. I'll remove the ternary.